extensions/tokenizers/src/main/java/ai/djl/huggingface/tokenizers/Encoding.java

```diff
@@ -55,23 +56,85 @@ protected Encoding(
         this.overflowing = overflowing;
     }
 
+    /**
+     * Returns the {@link NDList} representation of the encodings.
+     *
+     * @param encodings the {@code Encoding} batch
+     * @param manager the {@link NDManager} to create the NDList
+     * @param withTokenType true to include the token type id
+     * @param int32 true to use int32 datatype
+     * @return the {@link NDList}
+     */
+    public static NDList toNDList(
+            Encoding[] encodings, NDManager manager, boolean withTokenType, boolean int32) {
+        NDList list = new NDList();
+        if (!int32) {
+            long[][] ids = new long[encodings.length][];
+            long[][] attentionMask = new long[encodings.length][];
+            long[][] typeIds = new long[encodings.length][];
+            for (int i = 0; i < encodings.length; i++) {
+                ids[i] = encodings[i].getIds();
+                attentionMask[i] = encodings[i].getAttentionMask();
+                if (withTokenType) {
+                    typeIds[i] = encodings[i].getTypeIds();
+                }
+            }
+            list.add(manager.create(ids));
+            NDArray inputAttentionMask = manager.create(attentionMask);
+            list.add(inputAttentionMask);
+            if (withTokenType) {
+                list.add(manager.create(typeIds));
+            }
+            return list;
+        }
+
+        int[][] ids = new int[encodings.length][];
+        int[][] attentionMask = new int[encodings.length][];
+        int[][] typeIds = new int[encodings.length][];
+        for (int i = 0; i < encodings.length; i++) {
+            ids[i] = Arrays.stream(encodings[i].getIds()).mapToInt(l -> (int) l).toArray();
+            attentionMask[i] =
+                    Arrays.stream(encodings[i].getAttentionMask()).mapToInt(l -> (int) l).toArray();
+            if (withTokenType) {
+                typeIds[i] =
+                        Arrays.stream(encodings[i].getTypeIds()).mapToInt(l -> (int) l).toArray();
+            }
```

Comment:
This looks like the only part of this function that differs based on dtype -> can we refactor so only the logic in the the loop differs?

Reply:
the variable data type are different (`ids`, `attentionMask`, `typeIds`), they has to be declared separately. I'm not sure if there is easy way to combine them in the loop

Reply:
oh yea i didn't catch that. makes sense